baseplate/lib/message_queue.py

```diff
@@ -111,14 +147,100 @@ def unlink(self) -> None:
         self.queue.unlink()
 
     def close(self) -> None:
-        """Close the queue, freeing related resources.
+        self.queue.close()
 
-        This must be called explicitly if queues are created/destroyed on the
-        fly. It is not automatically called when the object is reclaimed by
-        Python.
 
-        """
-        self.queue.close()
+class InMemoryMessageQueue(MessageQueue):
+    """An in-memory inter process message queue.
+
+    Uses a simple Python Queue to store data.
+
+    Used in conjunction with the RemoteMessageQueue to
+    provide an alternative to Posix queues, for systems
+    that don't have Posix available. The client will
+    instantitate a RemoteMessageQueue, which connect to
+    a Thrift server. The Thrift server internally uses an
```

Comment:
This is probably a little late to ask, but why are we using Thrift here at all?  We, as a company, are trying to move away from Thrift.  Adding new uses of it seems incorrect.  Is there a design doc for this change anywhere at this point?  It seems to have gone in a very different direction than I would have expected when the PR was originally created.

Reply:
There's this one-pager: https://docs.google.com/document/d/1soN0UP9P12u3ByUwH_t47Uw9GwdVZRDuRFT0MvR52Dk/edit

Reply:
> This is probably a little late to ask, but why are we using Thrift here at all? We, as a company, are trying to move away from Thrift.

I did not think the grpc alternatives were ready for use or in baseplate yet. I'd be happy to hear the alternatives you had in mind.